// personalized_api/src/model/PAURLMAPPINGVIEWMODEL.js
/**
 * PersonalizedApi
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.PersonalizedApi) {
      root.PersonalizedApi = {};
    }
    root.PersonalizedApi.PAURLMAPPINGVIEWMODEL = factory(root.PersonalizedApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The PAURLMAPPINGVIEWMODEL model module.
   * @module model/PAURLMAPPINGVIEWMODEL
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>PAURLMAPPINGVIEWMODEL</code>.
   * @alias module:model/PAURLMAPPINGVIEWMODEL
   * @class
   */
  var exports = function() {
    var _this = this;












  };

  /**
   * Constructs a <code>PAURLMAPPINGVIEWMODEL</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PAURLMAPPINGVIEWMODEL} obj Optional instance to populate.
   * @return {module:model/PAURLMAPPINGVIEWMODEL} The populated <code>PAURLMAPPINGVIEWMODEL</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('ID')) {
        obj['ID'] = ApiClient.convertToType(data['ID'], 'Number');
      }
      if (data.hasOwnProperty('API_SET')) {
        obj['API_SET'] = ApiClient.convertToType(data['API_SET'], 'String');
      }
      if (data.hasOwnProperty('URL_MAPPING')) {
        obj['URL_MAPPING'] = ApiClient.convertToType(data['URL_MAPPING'], 'String');
      }
      if (data.hasOwnProperty('AUTHENTICATION_TYPE')) {
        obj['AUTHENTICATION_TYPE'] = ApiClient.convertToType(data['AUTHENTICATION_TYPE'], 'String');
      }
      if (data.hasOwnProperty('AUTHENTICATION_TYPEID')) {
        obj['AUTHENTICATION_TYPEID'] = ApiClient.convertToType(data['AUTHENTICATION_TYPEID'], 'Number');
      }
      if (data.hasOwnProperty('USERNAME')) {
        obj['USERNAME'] = ApiClient.convertToType(data['USERNAME'], 'String');
      }
      if (data.hasOwnProperty('PASSWORD')) {
        obj['PASSWORD'] = ApiClient.convertToType(data['PASSWORD'], 'String');
      }
      if (data.hasOwnProperty('TOKEN')) {
        obj['TOKEN'] = ApiClient.convertToType(data['TOKEN'], 'String');
      }
      if (data.hasOwnProperty('API_NAME')) {
        obj['API_NAME'] = ApiClient.convertToType(data['API_NAME'], 'String');
      }
      if (data.hasOwnProperty('DESCRIPTION')) {
        obj['DESCRIPTION'] = ApiClient.convertToType(data['DESCRIPTION'], 'String');
      }
      if (data.hasOwnProperty('VERSION')) {
        obj['VERSION'] = ApiClient.convertToType(data['VERSION'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {Number} ID
   */
  exports.prototype['ID'] = undefined;
  /**
   * @member {String} API_SET
   */
  exports.prototype['API_SET'] = undefined;
  /**
   * @member {String} URL_MAPPING
   */
  exports.prototype['URL_MAPPING'] = undefined;
  /**
   * @member {String} AUTHENTICATION_TYPE
   */
  exports.prototype['AUTHENTICATION_TYPE'] = undefined;
  /**
   * @member {Number} AUTHENTICATION_TYPEID
   */
  exports.prototype['AUTHENTICATION_TYPEID'] = undefined;
  /**
   * @member {String} USERNAME
   */
  exports.prototype['USERNAME'] = undefined;
  /**
   * @member {String} PASSWORD
   */
  exports.prototype['PASSWORD'] = undefined;
  /**
   * @member {String} TOKEN
   */
  exports.prototype['TOKEN'] = undefined;
  /**
   * @member {String} API_NAME
   */
  exports.prototype['API_NAME'] = undefined;
  /**
   * @member {String} DESCRIPTION
   */
  exports.prototype['DESCRIPTION'] = undefined;
  /**
   * @member {String} VERSION
   */
  exports.prototype['VERSION'] = undefined;



  return exports;
}));
