// personalized_api/src/model/UMUSERATTRIBUTEVIEWMODEL.js
/**
 * PersonalizedApi
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.PersonalizedApi) {
      root.PersonalizedApi = {};
    }
    root.PersonalizedApi.UMUSERATTRIBUTEVIEWMODEL = factory(root.PersonalizedApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The UMUSERATTRIBUTEVIEWMODEL model module.
   * @module model/UMUSERATTRIBUTEVIEWMODEL
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>UMUSERATTRIBUTEVIEWMODEL</code>.
   * @alias module:model/UMUSERATTRIBUTEVIEWMODEL
   * @class
   */
  var exports = function() {
    var _this = this;










  };

  /**
   * Constructs a <code>UMUSERATTRIBUTEVIEWMODEL</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/UMUSERATTRIBUTEVIEWMODEL} obj Optional instance to populate.
   * @return {module:model/UMUSERATTRIBUTEVIEWMODEL} The populated <code>UMUSERATTRIBUTEVIEWMODEL</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('cn')) {
        obj['cn'] = ApiClient.convertToType(data['cn'], 'String');
      }
      if (data.hasOwnProperty('country')) {
        obj['country'] = ApiClient.convertToType(data['country'], 'String');
      }
      if (data.hasOwnProperty('dateOfBirth')) {
        obj['dateOfBirth'] = ApiClient.convertToType(data['dateOfBirth'], 'String');
      }
      if (data.hasOwnProperty('gender')) {
        obj['gender'] = ApiClient.convertToType(data['gender'], 'String');
      }
      if (data.hasOwnProperty('givenName')) {
        obj['givenName'] = ApiClient.convertToType(data['givenName'], 'String');
      }
      if (data.hasOwnProperty('mail')) {
        obj['mail'] = ApiClient.convertToType(data['mail'], 'String');
      }
      if (data.hasOwnProperty('mobile')) {
        obj['mobile'] = ApiClient.convertToType(data['mobile'], 'String');
      }
      if (data.hasOwnProperty('profileConfiguration')) {
        obj['profileConfiguration'] = ApiClient.convertToType(data['profileConfiguration'], 'String');
      }
      if (data.hasOwnProperty('sn')) {
        obj['sn'] = ApiClient.convertToType(data['sn'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {String} cn
   */
  exports.prototype['cn'] = undefined;
  /**
   * @member {String} country
   */
  exports.prototype['country'] = undefined;
  /**
   * @member {String} dateOfBirth
   */
  exports.prototype['dateOfBirth'] = undefined;
  /**
   * @member {String} gender
   */
  exports.prototype['gender'] = undefined;
  /**
   * @member {String} givenName
   */
  exports.prototype['givenName'] = undefined;
  /**
   * @member {String} mail
   */
  exports.prototype['mail'] = undefined;
  /**
   * @member {String} mobile
   */
  exports.prototype['mobile'] = undefined;
  /**
   * @member {String} profileConfiguration
   */
  exports.prototype['profileConfiguration'] = undefined;
  /**
   * @member {String} sn
   */
  exports.prototype['sn'] = undefined;



  return exports;
}));
